homeassistant/components/xiaomi_tv/media_player.py

```diff
@@ -67,18 +74,44 @@ def name(self):
     @property
     def state(self):
         """Return _state variable, containing the appropriate constant."""
+        if self._tv.assume_state:
+            return self._state
+        if self._tv.is_on:
```

Comment:
This potentially does a request. That's not allowed in entity state properties since they will be evaluated in the event loop.

Please update `self._state` in `update` method.

Reply:
This still needs to move.

Reply:
And where can I check whether or not the TV is on or off?

Reply:
Inside `update` entity method.